templates/.github/workflows/prod-pipeline.yml

```diff
@@ -0,0 +1,78 @@
+# .github/workflows/prod-pipeline.yml
+# This is a github action pipeline for production.
+name: Production Deployment
+on: 
+  workflow_dispatch:
+    inputs:
+      git-ref:
+        description: Git Ref (Optional)    
+        required: false
+# Jobs
+jobs:
+  build-prod:
+      runs-on: ubuntu-latest
+      env:
+        CI: true
+        build_env: production
+      steps:
+      # Checkout the code
+        - uses: actions/checkout@v2
+      # Run Tests
+        - name: Run Unit Tests
+          uses: actions/setup-node@v1
+          with: 
+            node-version: 12.x
+        - run: |
+            npm install
+            npm run test
+      # Build Code
```

Comment:
The production pipeline shouldn't have a build step. Both staging and production should share an artifact.

Reply:
Hmm 🤔, K so I was thinking that because it's possible to have diffrent envs for both `prod` and `stage` they both need to have separate builds?

Reply:
In general during a build pipeline the goal should be to make an immutable, reusable artifact that doesn't have any state or configuration baked into it, then the environment supplies config at runtime. If you are generating different artifacts for each environment you have less confidence that your application is going to work the same on production as it does on staging.